ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -1145,6 +1110,142 @@ public void shutdown() throws StoreException {
+        replicaSealStatus.get() != ReplicaSealStatus.PARTIALLY_SEALED || (replicaStatusDelegates.size() > 1
+            && replicaSealStatus.getAndSet(ReplicaSealStatus.PARTIALLY_SEALED)
+            != ReplicaSealStatus.PARTIALLY_SEALED))) {
+      for (ReplicaStatusDelegate replicaStatusDelegate : replicaStatusDelegates) {
+        if (!replicaStatusDelegate.partialSeal(replicaId)) {
+          metrics.partialSealSetError.inc();
+          logger.warn("Could not set the partition as partially-sealed status on {}", replicaId);
+          replicaSealStatus.set(replicaSealStatus1);
+        } else {
+          metrics.partialSealDoneCount.inc();
+          logger.info(
+              "Store is successfully partial-sealed for partition : {} because current used capacity : {} bytes is exceeds Partial Read threshold : (low : {}, high : {}) bytes",
+              replicaId.getPartitionId(), index.getLogUsedCapacity(), partialSealThresholdBytesLow,
+              partialSealThresholdBytesHigh);
+        }
+      }
+    } else if (resolveReplicaSealStatusFromLogSize() == ReplicaSealStatus.NOT_SEALED && (
+        replicaSealStatus.get() != ReplicaSealStatus.NOT_SEALED || (replicaStatusDelegates.size() > 1
+            && replicaSealStatus.getAndSet(ReplicaSealStatus.NOT_SEALED) != ReplicaSealStatus.NOT_SEALED))) {
+      for (ReplicaStatusDelegate replicaStatusDelegate : replicaStatusDelegates) {
+        if (!replicaStatusDelegate.unseal(replicaId)) {
+          metrics.unsealSetError.inc();
+          logger.warn("Could not set the partition as not sealed on {}", replicaId);
+          replicaSealStatus.set(replicaSealStatus1);
+        } else {
+          metrics.unsealDoneCount.inc();
+          logger.info(
+              "Store is successfully unsealed for partition : {} because current used capacity : {} bytes is below Read Write threshold : {} bytes",
+              replicaId.getPartitionId(), index.getLogUsedCapacity(), partialSealThresholdBytesHigh);
+        }
+      }
+    }
+  }
+
+  /**
+   * Reconcile the sealed status of the replica during startup.
+   */
+  private void reconcileSealedStatus() {
+    // During startup, we also need to reconcile the replica state from both ZK clusters.
+    // reconcile the state by reading sealing state from both clusters
+    ReplicaSealStatus sealStatus = resolveReplicaSealStatusFromLogSize();
+    String partitionName = replicaId.getPartitionId().toPathString();
+    for (ReplicaStatusDelegate replicaStatusDelegate : replicaStatusDelegates) {
+      Set<String> sealedReplicas = new HashSet<>(replicaStatusDelegate.getSealedReplicas());
+      Set<String> partiallySealedReplicas = new HashSet<>(replicaStatusDelegate.getPartiallySealedReplicas());
+      ReplicaSealStatus sealStatus1 = sealedReplicas.contains(partitionName) ? ReplicaSealStatus.SEALED : (partiallySealedReplicas.contains(partitionName) ? ReplicaSealStatus.PARTIALLY_SEALED : ReplicaSealStatus.NOT_SEALED);
```

Comment:
nit: would it be okay to rename `sealStatus1` to `sealStatus` and `sealStatus` to `globalSealStatus` or something like that?